handlers/validator.go

```diff
@@ -609,9 +609,9 @@ func Validator(w http.ResponseWriter, r *http.Request) {
 			}
 
 			// add attestationStats that are not yet in validator_stats
-			lookback := int64(lastFinalizedEpoch - (lastStatsDay+1)*utils.EpochsPerDay())
-			if lookback > 0 {
-				missedAttestations, err := db.BigtableClient.GetValidatorMissedAttestationHistory([]uint64{index}, lastFinalizedEpoch-uint64(lookback), lastFinalizedEpoch)
+			lookback := int64(validatorPageData.Epoch - (lastStatsDay+1)*utils.EpochsPerDay())
+			if lookback > 0 && validatorPageData.Epoch > 0 {
```

Comment:
```suggestion
			nextDayFirstEpoch, _ := utils.GetFirstAndLastEpochForDay(lastStatsDay + 1)
			if validatorPageData.Epoch > nextDayFirstEpoch {
				lookback := ... // define as uint64
```

If you don't like this, consider still using `utils.GetFirstAndLastEpochForDay` instead of calculating `(lastStatsDay+1)*utils.EpochsPerDay()`

Reply:
By the way, while looking at this I have found another potential improvement, see 7e6d4f43060236fec1e84f0e981dbc19e96599f7

ptal